app/src/main/java/com/infomaniak/drive/ui/fileList/FileListViewModel.kt

```diff
@@ -163,18 +163,21 @@ class FileListViewModel(application: Application) : AndroidViewModel(application
         }
     }
 
-    fun getFavoriteFiles(order: SortType, isNewSort: Boolean): LiveData<FolderFilesResult?> {
+    fun getFavoriteFiles(order: SortType, isNewSort: Boolean, userDrive: UserDrive): LiveData<FolderFilesResult?> {
         getFilesJob.cancel()
         getFilesJob = Job()
         return liveData(Dispatchers.IO + getFilesJob) {
             tailrec suspend fun recursive(isFirstPage: Boolean, isNewSort: Boolean, cursor: String? = null) {
                 getFilesJob.ensureActive()
-                val apiResponse = ApiRepository.getFavoriteFiles(AccountUtils.currentDriveId, order, cursor)
+                val okHttpClient = AccountUtils.getHttpClient(userDrive.userId)
+                val apiResponse = ApiRepository.getFavoriteFiles(userDrive.driveId, order, cursor, okHttpClient)
                 if (apiResponse.isSuccess()) {
                     when {
                         apiResponse.data.isNullOrEmpty() -> emit(null)
                         apiResponse.hasMoreAndCursorExists -> {
-                            apiResponse.data?.let { FileController.saveFavoritesFiles(it, isFirstPage) }
+                            apiResponse.data?.let {
+                                saveFavoriteFiles(it, isFirstPage, userDrive)
```

Comment:
You can keep "FileController." here, it's easier to know from which controller the method is